ff/serving/provider/postgres.go

```diff
@@ -212,5 +238,46 @@ func (q postgresSQLQueries) numRows(n interface{}) (int64, error) {
 }
 
 func (q postgresSQLQueries) transformationCreate(name string, query string) string {
-	return fmt.Sprintf("CREATE TABLE %s AS %s ", sanitize(name), query)
+	return fmt.Sprintf("CREATE TABLE  %s AS %s", sanitize(name), query)
+}
+
+func (q postgresSQLQueries) transformationUpdate(db *sql.DB, tableName string, query string) error {
+	tempName := sanitize(fmt.Sprintf("tmp_%s", tableName))
+	fullQuery := fmt.Sprintf("CREATE TABLE %s AS %s", tempName, query)
+	return q.atomicUpdate(db, tableName, tempName, fullQuery)
+}
+
+func (q postgresSQLQueries) atomicUpdate(db *sql.DB, tableName string, tempName string, query string) error {
+	santizedName := sanitize(tableName)
+	oldName := sanitize(fmt.Sprintf("old_%s", tableName))
+	transaction := fmt.Sprintf("BEGIN;"+
+		"%s;"+
+		"ALTER TABLE %s RENAME TO %s;"+
+		"ALTER TABLE %s RENAME TO %s;"+
+		"DROP TABLE %s;"+
+		"COMMIT;", query, santizedName, oldName, tempName, santizedName, oldName)
+	_, err := db.Exec(transaction)
```

Comment:
If this fails, does the temp table persist in the database? Is there a way to delete and restart before the transaction?

Reply:
All actions are rollback automatically if one step fails, so it will just leave it will the original table and no temp table